src/index.ts

```diff
@@ -220,21 +245,33 @@ export const replacer = function replacer(options: Options) {
         return '[Object]';
       }
 
+      if (value === this) {
+        return `_duplicate_${JSON.stringify(keys)}`;
+      }
+
+      // when it's a class and we don't want to support classes, skip
+      if (
+        value.constructor &&
+        value.constructor.name &&
+        value.constructor.name !== 'Object' &&
+        !Array.isArray(value) &&
+        !options.allowClass
+      ) {
+        return undefined;
+      }
+
       const found = objects.get(value);
       if (!found) {
+        const converted = Array.isArray(value) ? value : convertUnconventionalData(value);
+
         if (
-          value &&
-          isObject(value) &&
           value.constructor &&
           value.constructor.name &&
-          value.constructor.name !== 'Object'
+          value.constructor.name !== 'Object' &&
+          !Array.isArray(value)
         ) {
-          if (!options.allowClass) {
```

Comment:
Regression?

Reply:
not really, but I added a line, to make this more obvious